### PR TITLE
fix the issue where searching entries by `entry_id` in CP did not wor…

### PR DESCRIPTION
**Pull request by intoeetive**
…k when entering less than 3 digits

EECORE-1593

### CODE REVIEW

File: system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php [hunk 1]
```diff
@@ -345,7 +345,11 @@ protected function setupEntries()
                     break;
             }
 
-            $entries->search($search_fields, $this->search_value);
+            if (is_numeric($this->search_value) && strlen($this->search_value) < 3) {
```
Comment (matthewjohns0n) [inline system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php:348]: So will this fix only work for entry IDs less than 100?

Comment (intoeetive) [reply system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php:348]: If it's greater than 999 the regular search will work (in all fields), as it was previously

Comment (bryannielsen) [reply system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php:348]: This is strlen < 3 so less than 100 unless this is an error and we want `strlen <= 3`

Comment (intoeetive) [reply system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php:348]: No, with 3 character we already use "normal search"

Comment (intoeetive) [reply system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php:348]: Previously this was based on rule set up in javascript `>= 3`
Now we also want to expand search to numbers `<3`

So if the search keyword is integer, we accept any number of characters, and search by entry ID only. If it's `>=3` (so `! < 3`) we search in all fields

**Review by matthewjohns0n — APPROVED** [review]

**Pull request merged**